tools/gen_vmlinux.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+set -e
+
+btf_dir=/sys/kernel/btf
+modules="\
+	openvswitch \
+	nf_tables \
+"
+
+# First load required modules
+for mod in $modules; do
+	modprobe $mod
+done
+
+tmpdir=$(mktemp -d)
+
+# Generate vmlinux header.
+bpftool btf dump file $btf_dir/vmlinux format c > $tmpdir/vmlinux.h
```

Comment:
an interesting idea discussed with and suggested by @amorenoz in order to generate a succinct but complete vmlinux.h was to have a list of types we need and check if we could generate a compileable header using `pahole`.
Not sure this is feasible (requires `pahole` to include the definition of all the types referenced in the "outer definitions"), but at least it would be interesting to investigate.
WDYT?

Reply:
It might be feasible but I'm wondering:
- Would this approach have any benefit compared to the one in this PR? Aka. would the added complexity be justified? From a compilation point of view, having more types than needed does not result in a different output.
- Wouldn't only including types we use prevent us from using new types easily? Given that generating `vmlinux.h` is not that trivial especially.

Reply:
> It might be feasible but I'm wondering:
> 
>     * Would this approach have any benefit compared to the one in this PR? Aka. would the added complexity be justified? From a compilation point of view, having more types than needed does not result in a different output.
> 

not really. That's a good point, though. We wanted to update our `vmlinux.h` because the one generated a while ago was fine initially, but it was supposed to be shrunk as it carried too many useless definitions.
Now it shrunk significantly, so I guess it can be considered ok. Still, the idea remains a good one, but maybe we can consider it for the future (for the time being this approach was and is fine).

>     * Wouldn't only including types we use prevent us from using new types easily? Given that generating `vmlinux.h` is not that trivial especially.

That's a possibility, yes. It's hard to tell how likely, I would guess not very.
As we said, if no strong reasons arise, we can keep the current approach, AFAIC.